src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystem.java

```diff
@@ -47,6 +57,67 @@ private static void createBusEquations(LfNetwork network, VariableSet variableSe
         }
     }
 
+    private static void createLocalVoltageEquation(VariableSet variableSet, EquationSystem equationSystem, LfBus bus) {
+        equationSystem.createEquation(bus.getNum(), EquationType.BUS_V).addTerm(new BusVoltageEquationTerm(bus, variableSet));
+        equationSystem.createEquation(bus.getNum(), EquationType.BUS_Q).setActive(false);
+    }
+
+    private static void createRemoteVoltageEquations(LfBus bus, List<LfBus> sourceBuses, EquationSystem equationSystem, VariableSet variableSet) {
```

Comment:
This equation system for remote control does not work on my use cases, from the first NR iteration. It meens that the issue is in the equation system, not in the reactive limits outer loop. I have tried to make something simpler: in my code test, I do as if the generator with remote control is directly connected at the remote bus and the source buses just have the reactive linked equations. 
````
equationSystem.createEquation(bus.getNum(), EquationType.BUS_REMOTE_V).addTerm(new BusVoltageEquationTerm(bus, variableSet)); // New PV bus.
equationSystem.createEquation(bus.getNum(), EquationType.BUS_Q).setActive(false); // Not a PQ bus anymore.

….

for (LfBus sourceBus : sourceBuses) {
            // deactivate reactive equation for all remote voltage source bus (where the generator is connected)
            // equationSystem.createEquation(sourceBus.getNum(), EquationType.BUS_Q).setActive(false);
           // Source buses are in fact PQ buses., but the sum of reactive power of source buses should be linked.  
        }
````

Reply:
But maybe it is not the equation system you have in mind…